mods/cnc/maps/scb30ea/scb30ea.lua

```diff
@@ -0,0 +1,376 @@
+		end)
+		Trigger.OnEnteredFootprint(CivsGDIHelpCellTrigger, function(a, id)
+			if a == RunForHelpCivs[1] or a == RunForHelpCivs[2] then
+				Reinforcements.Reinforce(GDI, CiviliansHelpTeam, { CPos.New(2,9), CPos.New(3,9) }, 30, function(a)
+					a.AttackMove(waypoint2.Location)
+					a.AttackMove(waypoint9.Location)
+					a.AttackMove(waypoint8.Location)
+					IdleHunt(a)
+				end)
+				Trigger.RemoveFootprintTrigger(id)
+			end
+		end)
+		CivsRunning = true
+		local cam = Actor.Create("camera", true, { Owner = Nod, Location = CPos.New(53,44) })
+		Trigger.AfterDelay(125, function() cam.Destroy() end)
+		Media.DisplayMessage(UserInterface.Translate("civilians-runs"))
+	end
+end
+
+
+CivsBuildingsToDestroy = 0
+CheckVillageDestruction = function()
+	CivsBuildingsToDestroy = CivsBuildingsToDestroy - 1
+	if CivsBuildingsToDestroy == 2 then
+		Media.DisplayMessage(UserInterface.Translate("village-destruction-warning"))
+	elseif CivsBuildingsToDestroy == 0 then
+		Reinforcements.Reinforce(GDI, ReinforcementsMammoths, { CPos.New(2,9), CPos.New(3,9) }, 40, function(a)
+			a.AttackMove(waypoint11.Location)
+			a.AttackMove(waypoint5.Location)
+			a.AttackMove(waypoint4.Location)
+			a.AttackMove(waypoint6.Location)
+			IdleHunt(a)
+		end)
+	end
+end
+
+
+GuardBase = function()
+	Utils.Do(GDIBase, function(building)
+		GuardBuilding(building)
+	end)
+end
+
+
+GuardBuilding = function(building)
+	Trigger.OnDamaged(building, function(slf, atk, dmg)
+		if atk.Type ~= "player" and not atk.IsDead and atk.Owner == Nod then
+			Utils.Do(BaseDefenseTeam, function(guard)
+				if not guard.IsDead and not building.IsDead then
+					if guard.Stance == "Defend" then
+						guard.Stop()
+						guard.Stance = "AttackAnything"
+						guard.AttackMove(atk.Location, 3)
+						Trigger.OnIdle(guard, function()
+							guard.AttackMove(waypoint12.Location, 3)
+							guard.Stance = "Defend"
+							Trigger.ClearAll(guard)
+						end)
+					end
+				end
+			end)
+		end
+	end)
+end
+
+
+StartGuard = function()
+	Trigger.OnAllKilled(BaseDefenseTeam, function()
+		table.insert(UniqueTeamsQueue, { team = DefenseTeams, job = "defend" })
+	end)
+	Utils.Do(BaseDefenseTeam, function(guard)
+		guard.Stance = "Defend"
+	end)
+end
+
+
+StartPatrol = function()
+	Trigger.OnAllKilled(PatrolTeam, function()
+		table.insert(UniqueTeamsQueue, { team = PatrolTeams, job = "patrol" })
+	end)
+	Utils.Do(PatrolTeam, function(a)
+		Trigger.OnKilled(a, function()
+			Utils.Do(PatrolTeam, function(a)
+				if not a.IsDead then
+					a.Stop()
+					IdleHunt(a)
+				end
+			end)
+		end)
+	end)
+	MoveAsGroup(PatrolTeam, PatrolPath, 1, true)
+end
+
+
+MoveAsGroup = function(team, path, i, loop)
+	if i == 1 and not loop then
+		Utils.Do(team, function(u)
+			Trigger.OnDamaged(u, function()
+				Trigger.AfterDelay(2, function()
+					Utils.Do(team, function(u)
+						if not u.IsDead then
+							Trigger.Clear(u, "OnDamaged")
+							IdleHunt(u)
+						end
+					end)
+				end)
```

Comment:
This delay seems harmless but also not necessary. If you were *adding* a trigger in the same tick, that could certainly be a problem. A similar thing is done further down with `OnIdle` (line 190-191), but that part has no delay. If this is a mistake on my part, please pardon me. It struck me as clutter in a function that already has a lot going on.